File: java-dsa/njasm/CyclicSortAlgo.java
package njasm;

import java.util.Arrays;

public class CyclicSortAlgo {
    public static void main(String[] args){
        int[] arr = {5,3,1,6,2,4};
        System.out.println(Arrays.toString(SortArrayCyclic(arr)));
    }

   static int[] SortArrayCyclic(int[] arr) {
        for (int i = 0; i < arr.length;){
            if (arr[i] != i+1){
                int temp = arr[i];
                arr[i] = arr[temp-1];
                arr[temp-1] = temp;
            } else{
                i++;
            }
        }
        return arr;
    }

    
}

File: golang-dsa/go-dsa/cyclic_sort_algo.go
package app

func CyclicSortAlgo(arr []int) { // {1, 3, 4, 6, 5, 2}
	i := 0
	for i < len(arr)-1 {
		if arr[i] != i+1 {
			temp := arr[i] - 1
			SwapTwo(arr, temp, i)
		} else {
			i++
		}
	}
}

func SwapTwo(arr []int, first int, second int) {
	temp := arr[first]
	arr[first] = arr[second]
	arr[second] = temp

}
